rasp/tor_engine.py

```diff
@@ -1,12 +1,167 @@
 import getpass
 import os
 from contextlib import contextmanager
+from datetime import datetime, timedelta
 
 from stem import Signal
 from stem.connection import connect
 
 from rasp.base import DefaultEngine
-from rasp.errors import EngineError
+from rasp.errors import ControllerError
+
+
+class TorController(object):
+    """
+    Attributes:
+        address (str): IP address of Tor SOCKS proxy to connect,
+            can also be set with the ``RASP_TOR_ADDRESS``
+            environment variable.
+        control_port (int, optional): Port number for control
+            port usage to refresh endpoint address,
+            can also be set with the ``RASP_TOR_CONTROL_PASSWORD``
+            environment variable.
+        control_password (str, optional): Password to protect control
+            port usage,
+            can also be set with the ``RASP_TOR_CONTROL_PORT``
+            environment variable.
+        callback (callable, optional): A function that returns a boolean value
+            to determine when to send a signal to Tor"""
+
+    def __init__(self,
+                 address=None,
+                 port=None,
+                 password=None,
+                 signal_limiter=None):
+        self.address = (
+            address
+            or os.environ.get('RASP_TOR_ADDRESS')
```

Comment:
Yeah I've been working with the Ops folks at my company a lot and they're huge fans of this design.  The way it was explained to me is that all configuration for "outside" code/processes should happen outside of the program.